samples/cpp/string_support_test.cpp

```diff
@@ -0,0 +1,161 @@
+        constructor_validate_and_infer_types();
+    }
+
+    void validate_and_infer_types() override {
+        // The conditions below is a part of CPU hack to support a wrapped string tensor as a u8 tensor with a pointer
+        if(get_input_size() == 1 && get_input_element_type(0) == ov::element::u8) {
+            std::cerr << "[ DEBUG ] StringTensorOp::validate_and_infer_types detected CPU hack at input, still providing element::string in the output\n";
+        }
+        if(get_output_size() == 1 && get_output_element_type(0) == ov::element::u8) {
+            std::cerr << "[ DEBUG ] StringTensorOp::validate_and_infer_types detected CPU hack at output, not overriding output type\n";
+            return;
+        }
+        // Default behviour without applied hack at the output
+        set_output_type(0, ov::element::string, get_input_partial_shape(0));
+    }
+
+    std::shared_ptr<ov::Node> clone_with_new_inputs(const ov::OutputVector& inputs) const override {
+        auto result = std::make_shared<StringTensorOp>(inputs);
+        return result;
+    }
+
+    bool visit_attributes(ov::AttributeVisitor& visitor) override {
+        return true;
+    }
+
+    bool has_evaluate() const {
+        return true;
+    }
+
+    bool evaluate(ov::TensorVector& outputs, const ov::TensorVector& inputs) const {
+        std::cerr << "Input raw tensor type: " << inputs[0].get_element_type() << " and shape " << inputs[0].get_shape() << "\n";
+
+        auto pitensor = &inputs[0];
+
+        if(pitensor->get_element_type() == ov::element::u8 && pitensor->get_byte_size() == sizeof(void*)) {
+            std::cerr << "[ DEBUG ] StringTensorOp::evaluate detected CPU hack at input, unwrapping\n";
+            auto data = *reinterpret_cast<const void* const*>(pitensor->data());
+            if(data != nullptr) {
+                pitensor = reinterpret_cast<const ov::Tensor*>(data);
+            }
+        }
+
+        auto input_tensor = *pitensor;
+
+        std::cerr << "Input tensor type after optional unwrapping: " << input_tensor.get_element_type() << " and shape " << input_tensor.get_shape() << "\n";
+        std::cerr << "Output raw tensor type: " << outputs[0].get_element_type() << " and shape " << outputs[0].get_shape() << "\n";
+
+        auto potensor = &outputs[0];
+
+        if(potensor->get_element_type() == ov::element::u8 && potensor->get_byte_size() == sizeof(void*)) {
+            std::cerr << "[ DEBUG ] StringTensorOp::evaluate detected CPU hack at output, unwrapping\n";
+            auto data = *reinterpret_cast<void* const*>(potensor->data());
+            if(data != nullptr) {
+                potensor = reinterpret_cast<ov::Tensor*>(data);
+            }
+        }
+
+        auto output_tensor = *potensor;
+
+        std::cerr << "Output tensor type after optional unwrapping: " << output_tensor.get_element_type() << " and shape " << output_tensor.get_shape() << "\n";
+        output_tensor.set_shape(input_tensor.get_shape());
```

Comment:
Part of the CPU hack: shouldn't be required as validate_infer_types will bring correctly computed shape and I believe CPU plugin should provide correctly shaped tensors for outputs.